daemon/container_unix.go

```diff
@@ -976,9 +987,9 @@ func (container *Container) connectToNetwork(idOrName string, updateSettings boo
 		if err != nil {
 			return err
 		}
-	}
 
-	container.updateSandboxNetworkSettings(sb)
+		container.updateSandboxNetworkSettings(sb)
```

Comment:
It should not happen like that. Only the first time when WalkSandboxes yields should the sandbox Id be set in the NetworkSettings. Not for every connect.